src/phoneField/PhoneField.tsx

```diff
@@ -0,0 +1,266 @@
+import React, { PureComponent, RefObject } from 'react'
+import cc from 'classcat'
+import prefix from '_utils'
+import isEmpty from 'lodash.isempty'
+import isEqual from 'lodash.isequal'
+import SelectField from 'selectField'
+import TextField, { inputTypes } from 'textField'
+import { allCountries } from 'country-telephone-data'
+
+export type selectfield = HTMLSelectElement
+
+enum FIELDS {
+  PHONENUMBER = 'phoneNumber',
+  PHONEREGION = 'phoneRegion',
+}
+interface PhoneFieldOnChangeParameters {
+  name: string
+  value: {
+    [FIELDS.PHONENUMBER]: string
+    [FIELDS.PHONEREGION]: string
+    phonePrefix: string
+    completePhoneNumber: string
+  }
+}
+
+type errorField = string | JSX.Element
+interface PhoneFieldProps {
+  readonly name: string
+  readonly onChange: (obj: PhoneFieldOnChangeParameters) => void
+  readonly id?: string
+  readonly className?: Classcat.Class
+  readonly extraClassName?: Classcat.Class
```

Comment:
extraClassName makes me think that is will be on the same element as className (i.e. the root).
Can we use a naming that tries to identify where the class is added ?
In that case, maybe innerWrapperClassName ?

Reply:
Done PTAL